### PR TITLE
Add support for large unicode grace files

### PR DESCRIPTION
**Pull request by IsaacOscar**
This should now support grace files larger than 65KB (the new limit is likely slightly less than 2GB).
Grace files can now also contain non-ASCII unicode characters.

### CODE REVIEW

File: src/som/compiler/KernanClient.java
Comment (smarr) [inline src/som/compiler/KernanClient.java]: Could the following lines be a loop to avoid some of the redundancy?

**Review by smarr — APPROVED** [review]
Not sure whether I am the best reviewer, but looks good, I think.

File: src/som/compiler/KernanClient.java
Comment (IsaacOscar) [reply src/som/compiler/KernanClient.java]: Sure if you want, see commit eafb0d5.

File: src/som/compiler/KernanClient.java
Comment (smarr) [inline src/som/compiler/KernanClient.java]: One more thing: this here is problematic. Swallowing exceptions like this is going to make for very unhappy users. For some reason, nothing happens if things go wrong. That's hard to diagnose.

I'd propose to wrap the exception into a RuntimeException instead. Then at least users will get a clue what's going wrong.

File: src/som/compiler/KernanClient.java
Comment (smarr) [inline src/som/compiler/KernanClient.java]: One more thing: this here is problematic. Swallowing exceptions like this is going to make for very unhappy users. For some reason, nothing happens if things go wrong. That's hard to diagnose.

I'd propose to wrap the exception into a RuntimeException instead. Then at least users will get a clue what's going wrong.

**Review by smarr — COMMENTED** [review]
One more things :)

Not sure whether you agreed with @richard-roberts or @kjx on how to do these things.
So, I am happy for you to hit the merge button yourself, I think.

File: src/som/compiler/KernanClient.java
Comment (IsaacOscar) [reply src/som/compiler/KernanClient.java]: That was the intention behind "assert false"; I beleive that said exception can never be thrown (since "UTF8" is a valid encoding); if I have made a mistake, the user should hopefully get a helpfull backtrace.

But I guess assertions can be turned off so your suggestion is probably better.

Comment (smarr) [reply src/som/compiler/KernanClient.java]: Yeah, assertions aren't reliable unfortunately.

File: src/som/compiler/KernanClient.java
Comment (IsaacOscar) [reply src/som/compiler/KernanClient.java]: > One more things :)
> 
> Not sure whether you agreed with @richard-roberts or @kjx on how to do these things.
> So, I am happy for you to hit the merge button yourself, I think.

No I've not had such a discussion, as I'm not really working on moth, rather I was just trying it out.

I know @Jamigrith is working on this branch, so I didn't want to disturb her work by making random changes.

Comment (IsaacOscar) [reply src/som/compiler/KernanClient.java]: Actually it seems I don't have write access to this repository, so I can't merge this myself.

Comment (smarr) [reply src/som/compiler/KernanClient.java]: Ok, then it's probably best to leave the merge to @Jamigrith :)

**Comment by ghost** [comment]
Thanks for the pull request @IsaacOscar and thanks for the helpful review @smarr. I think we should merge this, as I don't believe these changes will conflict with what @Jamigrith is working on.

**Pull request merged**
